common/rpc/middleware.go

```diff
@@ -47,3 +47,32 @@ func (m *authOutboundMiddleware) Call(ctx context.Context, request *transport.Re
 
 	return out.Call(ctx, request)
 }
+
+const _responseInfoContextKey = "response-info"
+
+// ContextWithResponseInfo will create a child context that has ResponseInfo set as value.
+// This value will get filled after the call is made and can be used later to retrieve some info of interest.
+func ContextWithResponseInfo(parent context.Context) (context.Context, *ResponseInfo) {
+	responseInfo := &ResponseInfo{}
+	return context.WithValue(parent, _responseInfoContextKey, responseInfo), responseInfo
+}
+
+// ResponseInfo structure is filled with data after the RPC call.
+// It can be obtained with rpc.ContextWithResponseInfo function.
+type ResponseInfo struct {
+	Size int
+}
+
+type responseInfoMiddleware struct{}
+
+func (m *responseInfoMiddleware) Call(ctx context.Context, request *transport.Request, out transport.UnaryOutbound) (*transport.Response, error) {
+	response, err := out.Call(ctx, request)
+
+	if value := ctx.Value(_responseInfoContextKey); value != nil {
```

Comment:
could we make this a public function so that we can get the response size in client?